### PR TITLE
Fix menu items that trigger secondary interface missing ellipsis

### LINKED ISSUE #80347: Minor Issue - Some Menu Items that Trigger a Secondary Interface are Missing an Ellipsis
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Godot v4.1.1.stable - Windows 10.0.22621 - Vulkan (Compatibility) - NVIDIA GeForce GTX 1060 (NVIDIA; 30.0.14.7311) - Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz (12 Threads)

### Issue description

OK, so this is minor, but having a bit of UI experience on Windows it is a bit of a _pet peeve_.   Most of Godot's menu items get this right, but some are missing it when they should have it, and most of the context menu items are definitely missing it.

To recap standard Windows UI behavior: if you have a menu item that when activated, raises a modal dialog that **_must be interacted with in order to complete the functionality indicated by the menu item_**, it should have an ellipsis at the end of it.  OK, so that is a mouthful so I will provide two examples of when you do and do not include the ellipsis.

When you right-click on a Node and select "**Add Child Node**" from the context menu, the menu item should really have an ellipsis and be shown as "**Add Child Node...**"  Why?  Because the user **must interact** with the following modal dialog in order to complete the functionality of _adding a child node_.  Also, they have the option to _cancel_ the operation entirely.

On the other hand, the ubiquitous **Help --> About** menu item generally does NOT get an ellipsis.  Why not?  Because once you have activated the menu item and shown the about box, the user does not have to do anything to complete the "about" functionality.  It is **_done_** - the showing of the _About_ dialog _completed_ the requested functionality.  The only thing they can do is dismiss the dialog.  Because the functionality is actually completed by activation of the menu item alone, there is no need for an ellipsis.

That said, most of the main menu items are correct, but there are a few like **Project --> Version Control -->***, whose subitems should likely have the ellipsis.  But quite a few of the context or pop-up menu items are missing them.

Also, right-clicking on an empty scene raises a menu with **Add Node Here** and **Instantiate Scene Here**, both of which should have an ellipsis on them.
 
On the context menu for the Scene pane, the context menu has a few items that are missing an ellipsis including the ones mentioned above as well as **Attach Script** and **Change Type**.

### Steps to reproduce

Open Godot and open the **Project --> Version Control** submenu.  Note the subitems are missing an ellipsis but should likely have them.

Create an empty 2D Scene and then right-click on the Node to raise the context menu.  The **Add Child Node**, **Instantiate Child Scene**, **Attach Script**, and **Change Type** should have an ellipsis because they all raise an interface that must be used to complete the functionality.

Right-click on the empty 2D view and note that **Add Node Here** and **Instantiate Scene Here** are missing an ellipsis because both commands raise an interface that must be used to complete the functionality.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by jcovin293**
This PR fixes #80347 where menu items that trigger a secondary interface (such as a dialog) do not have an ellipsis on their menu items.

### CODE REVIEW

File: editor/scene_tree_dock.cpp [hunk 1]
```diff
@@ -3761,14 +3761,14 @@ SceneTreeDock::SceneTreeDock(Node *p_scene_root, EditorSelection *p_editor_selec
...
+	ED_SHORTCUT("scene_tree/instantiate_scene", TTR("Instantiate Child Scene..."), KeyModifierMask::CMD_OR_CTRL | KeyModifierMask::SHIFT | Key::A);
 	ED_SHORTCUT("scene_tree/expand_collapse_all", TTR("Expand/Collapse Branch"));
 	ED_SHORTCUT("scene_tree/cut_node", TTR("Cut"), KeyModifierMask::CMD_OR_CTRL | Key::X);
 	ED_SHORTCUT("scene_tree/copy_node", TTR("Copy"), KeyModifierMask::CMD_OR_CTRL | Key::C);
 	ED_SHORTCUT("scene_tree/paste_node", TTR("Paste"), KeyModifierMask::CMD_OR_CTRL | Key::V);
-	ED_SHORTCUT("scene_tree/change_node_type", TTR("Change Type"));
-	ED_SHORTCUT("scene_tree/attach_script", TTR("Attach Script"));
+	ED_SHORTCUT("scene_tree/change_node_type", TTR("Change Type..."));
+	ED_SHORTCUT("scene_tree/attach_script", TTR("Attach Script..."));
 	ED_SHORTCUT("scene_tree/extend_script", TTR("Extend Script"));
```
Comment (Calinou) [inline editor/scene_tree_dock.cpp:3772]: I believe this one should have ellipsis too:

```suggestion
	ED_SHORTCUT("scene_tree/extend_script", TTR("Extend Script..."));
```

**Comment by akien-mga** [comment]
Could you amend the commit message to be more explicit? The PR title is good, the commit should use the same title (and can include "Fixes #80347" in the body of the commit message, not in the title).

**Comment by jcovin293** [comment]
I have amended the commit with the requested changes. Let me know if there are any other things that need to be added.

**Review by YuriSizov — APPROVED** [review]
Looks good to me, but needs a rebase.

**Comment by jcovin293** [comment]
I'm new to git. How would I rebase because there seems to be a merge conflict in ```scene_tree_dock.cpp```?

**Comment by jcovin293** [comment]
I use Visual Studio Code. I have tried resolving the conflict via the UI but it doesn't seem to be working as expected as it requires me to make another commit which might not look good on the git history.

**Comment by YuriSizov** [comment]
@jcovin293 While I'm not sure how it would create a new commit, but you can fix that after the fact by doing an interactive rebase and squashing your commits together with the `fixup` option.

**Comment by AThousandShips** [comment]
You accidentally closed your PR by resetting the branch, if you push to the branch again we can reopen

**Comment by jcovin293** [comment]
I have pushed a new commit. Do I need to do something to reopen it or is it on your end?

File: editor/scene_tree_dock.cpp [hunk 2]
```diff
@@ -3771,15 +3771,15 @@ SceneTreeDock::SceneTreeDock(Node *p_scene_root, EditorSelection *p_editor_selec
...
 	ED_SHORTCUT("scene_tree/expand_collapse_all", TTR("Expand/Collapse Branch"));
 	ED_SHORTCUT("scene_tree/cut_node", TTR("Cut"), KeyModifierMask::CMD_OR_CTRL | Key::X);
 	ED_SHORTCUT("scene_tree/copy_node", TTR("Copy"), KeyModifierMask::CMD_OR_CTRL | Key::C);
 	ED_SHORTCUT("scene_tree/paste_node", TTR("Paste"), KeyModifierMask::CMD_OR_CTRL | Key::V);
 	ED_SHORTCUT("scene_tree/paste_node_as_sibling", TTR("Paste as Sibling"), KeyModifierMask::CMD_OR_CTRL | KeyModifierMask::SHIFT | Key::V);
-	ED_SHORTCUT("scene_tree/change_node_type", TTR("Change Type"));
-	ED_SHORTCUT("scene_tree/attach_script", TTR("Attach Script"));
+	ED_SHORTCUT("scene_tree/change_node_type", TTR("Change Type..."));
+	ED_SHORTCUT("scene_tree/attach_script", TTR("Attach Script..."));
 	ED_SHORTCUT("scene_tree/extend_script", TTR("Extend Script"));
```
Comment (YuriSizov) [inline editor/scene_tree_dock.cpp:3783]: It seems you've lost the suggested change here.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**